plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -227,7 +252,12 @@ public CompletionStage<DeleteDataObjectResponse> deleteDataObjectAsync(DeleteDat
             .build();
         return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<DeleteDataObjectResponse>) () -> {
             try {
+                // TODO need to return SEQ_NO here
+                // If document doesn't exist, increment and return highest seq no ever seen, but we would have to track seqNo here
+                // If document never existed, return -2 (unassigned) for seq no (probably what we have to do here)
+                // If document exists, increment and return SEQ_NO
```

Comment:
In case of delete, why does seq no be updated? can we safely delete document as long as seq no matches?

Reply:
@arjunkumargiri This is somewhat of an artifact of how "soft deletes" work in OpenSearch. 

When you call the Delete API, it only marks the document as deleted in the metadata but it doesn't actually remove it until the segment is merged.   It's actually still available in search results [unless you use the `refresh` parameter](https://opensearch.org/docs/latest/api-reference/document-apis/delete-document/).

So it still tracks the `seq_no` as a change to the document for replication/tracking purposes until the document actually goes away on a merge, at which point I assume that tracking disappears.

Reply:
How can we distinguish between 'If the document doesn't exist' and 'If the document never existed'?